src/libsyntax/feature_gate.rs

```diff
@@ -820,7 +826,7 @@ impl<'a> Context<'a> {
             // feature gate checking. Macro gating runs
             // before the plugin attributes are registered
             // so we skip this then
-            if !is_macro {
+            if !is_macro && !attr::is_used(attr) {
```

Comment:
How come this part was modified as well? Presumably this means that we're not gating unused attributes now?